### PR TITLE
AIP-2698 - Adding code coverage to the Metaflow integration testing

### PR DESCRIPTION
**Pull request by hsezhiyan**
- Uses `pytest-cov` to report coverage
- Adds a gate of 65% coverage

Example coverage report: https://gitlab.zgtools.net/analytics/artificial-intelligence/ai-platform/aip-workflow/aip-metaflow-integration-testing/-/jobs/3714825

### CODE REVIEW

File: metaflow/plugins/kfp/tests/setup.cfg [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[coverage:run]
+omit = ./*
```
Comment (talebzeghmi) [inline metaflow/plugins/kfp/tests/setup.cfg:2]: how does this get coverage for what is under `/metaflow/metaflow/plugins/kfp`?

File: metaflow/plugins/kfp/tests/setup.cfg [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+[coverage:report]
+fail_under = 65
```
Comment (talebzeghmi) [inline metaflow/plugins/kfp/tests/setup.cfg:5]: Could you publish/copy the coverage HTML so we can view what isn't being covered?
```
-- Docs: https://docs.pytest.org/en/stable/warnings.html
----------- coverage: platform linux, python 3.6.8-final-0 -----------
Name                                                                 Stmts   Miss  Cover
----------------------------------------------------------------------------------------
/home/zservice/metaflow/metaflow/plugins/kfp/__init__.py                 0      0   100%
/home/zservice/metaflow/metaflow/plugins/kfp/kfp.py                    178     20    89%
/home/zservice/metaflow/metaflow/plugins/kfp/kfp_cli.py                 70      7    90%
/home/zservice/metaflow/metaflow/plugins/kfp/kfp_constants.py            8      0   100%
/home/zservice/metaflow/metaflow/plugins/kfp/kfp_decorator.py           43     24    44%
/home/zservice/metaflow/metaflow/plugins/kfp/kfp_foreach_splits.py      82     44    46%
/home/zservice/metaflow/metaflow/plugins/kfp/kfp_step_function.py       18     17     6%
/home/zservice/metaflow/metaflow/plugins/kfp/kfp_step_init.py           37     30    19%
----------------------------------------------------------------------------------------
TOTAL                                                                  436    142    67%
Required test coverage of 65.0% reached. Total coverage: 67.43%
```

File: metaflow/plugins/kfp/tests/setup.cfg (hunk 1 above)
Comment (hsezhiyan) [reply metaflow/plugins/kfp/tests/setup.cfg:2]: @talebzeghmi , it works as follows:
If you look at the changes in the `.gitlab-ci.yml` file, you'll see `--cov=../ --cov-config=setup.cfg`. The `--cov=../` indicates we'll provide coverage for the directory one level above from where the `.gitlab-ci.yml` file is located (`/metaflow/metaflow/plugins/kfp/tests` (`.gitlab-ci.yml` file is located) -> `/metaflow/metaflow/plugins/kfp`).
The `omit = ./*` indicates we will not provide coverage for any files within the same directory as where `setup.cfg` is located (`/metaflow/metaflow/plugins/kfp/tests`).

This prevents us from providing coverage from sample flows and integration testing code and focus only on kfp plugin code. Is that clear?

Comment (talebzeghmi) [reply metaflow/plugins/kfp/tests/setup.cfg:2]: Can the `--cov=../` be provided in setup.cfg next to the `omit=` statement?  That'd make it easier to find and follow

File: metaflow/plugins/kfp/tests/setup.cfg (hunk 2 above)
Comment (hsezhiyan) [reply metaflow/plugins/kfp/tests/setup.cfg:5]: @talebzeghmi where do you want to coverage HTML published?

Comment (talebzeghmi) [reply metaflow/plugins/kfp/tests/setup.cfg:5]: Checkout gitlab publish code coverage. You can find examples of how our cicd does it. It's a command to copy a directory

**Comment by hsezhiyan** [comment]
@talebzeghmi both your comments have been addressed. This is ready for review again.

**Review by talebzeghmi — APPROVED** [review]
thank you!

**Comment by hsezhiyan** [comment]
Correct publishing of code coverage will be completed in https://zbrt.atl.zillow.net/browse/AIP-2982.

**Pull request merged**